### ISSUE TITLE
FluentAssertion's "Subject" is not removed when matching a Dictionary

### ISSUE THREAD

**Issue by sgabler-solytic** [SwissLife-OSS/snapshooter#118]
**Describe the bug**
Hi, I ran into a similar issue as #56. I'm trying to use FluentAssertions with a custom object that looks like this:

`Dictionary<string,List<Dictionary<string,object>>`

The snapshot looks like this:

```
{
  "Subject": {
    "Customer": [
      {
        "ID": 1,
        "Name": "Bla",
        "EmailAddress": "blabla@blabla.com",
        "ContactId": null,
        "CreatedAt": "2021-01-10T19:03:59.075874+00:00",
        "LastUpdatedAt": "2021-01-10T19:03:59.075874+00:00"
      }
    ]
  }
}
```

I set a breakpoint in the `ObjectWrapperRemover` and get the namespace `FluentAssertions.Collections` for the dictionary. Since this namespace is not part of the if condition, the `Subject` is not removed.

https://github.com/SwissLife-OSS/snapshooter/blob/72098b23e88ae58497a733b3bc66ab325cac1fcd/src/Snapshooter/Helpers/ObjectWrapperRemover.cs#L18-L34

**To Reproduce**
1. Use FluentAssertions on a `Dictionary<string,List<Dictionary<string,object>>`
2. Check the snapshot and see that "Subject" is the top level field

**Expected behavior**
"Subject" is removed

Tried on Snapshooter 0.5.8

**Comment by huysentruitw** [comment]
I just had the same problem with a plain string object when changing from `Snapshot.Match(stringValue)` to `stringValue.Should().Match()`.

I am creating a PR to fix both.

**Comment by RohrerF** [comment]
Should be fixed in [v0.6.0](https://github.com/SwissLife-OSS/snapshooter/releases/tag/0.6.0) thanks to @huysentruitw 🥳
